datameta/api/groups.py

```diff
@@ -31,14 +31,38 @@ class GroupSubmissions:
 
     def __init__(self, request:Request, group_id:str):
         db = request.dbsession
-        # To Developer:
-        # Please insert code that queries the db
-        # for all submissions for the given group_id
-        # and store that list in self.submissions
-
-
-    def __json__(self) -> List[dict]:
-        return self(submissions)
+        group = resource_by_id(db, models.Group, group_id)
```

Comment:
This code may potentially yield a very high number of SQL queries, since SQLAlchemy lazily loads the relationship attributes as you access them. Since we can already anticipate here which relationships we're going to use, we can instruct SQLAlchemy to join everything in one go so that the subsequent access of the relationship attributes doesn't yield additional SQL queries. This will require additional imports of the model classes as well as
```python
from sqlalchemy.orm import joinedload
```

```suggestion
        group = resource_query_by_id(db, models.Group, group_id).options(
                joinedload(Group.submissions).joinedload(Submission.metadatasets).joinedload(MetaDataSet.metadatumrecords).joinedload(MetaDatumRecord.metadatum)
                ).one_or_none()
```